name.abuchen.portfolio/src/name/abuchen/portfolio/online/impl/PortfolioReportNet.java

```diff
@@ -209,77 +209,55 @@ else if (!symbol.equals(property.getValue()))
 
     public List<ResultItem> search(String query, SecuritySearchProvider.Type type) throws IOException
     {
-        try
-        {
-            URIBuilder uriBuilder = new URIBuilder().setScheme("https").setHost(HOST) //$NON-NLS-1$
-                            .setPath("/api/securities/search/" + query); //$NON-NLS-1$
-
-            if (type != null)
-            {
-                if (type == SecuritySearchProvider.Type.SHARE)
-                    uriBuilder.addParameter("type", TYPE_SHARE); //$NON-NLS-1$
-                else if (type == SecuritySearchProvider.Type.BOND)
-                    uriBuilder.addParameter("type", TYPE_BOND); //$NON-NLS-1$
-            }
-
-            URL searchUrl = uriBuilder.build().toURL();
+        WebAccess webAccess;
+        // return webAccess.getDocument();
 
-            URLConnection con = searchUrl.openConnection();
-            con.setConnectTimeout(1000);
-            con.setReadTimeout(5000);
-
-            return readItems(con);
+        if (type != null)
+        {
+            webAccess = WebAccess.builder().document("https", HOST, "/api/securities/search/" + query)//$NON-NLS-1$ //$NON-NLS-2$
+                            .withParameter(WebAccessParameter.builder().addParameter("type", //$NON-NLS-1$
+                                            type == SecuritySearchProvider.Type.SHARE ? TYPE_SHARE : TYPE_BOND))
+                            .build();
         }
-        catch (URISyntaxException e)
+        else
         {
-            throw new IOException(e);
+            webAccess = WebAccess.builder().document("https", HOST, "/api/securities/search/" + query)//$NON-NLS-1$ //$NON-NLS-2$
+                            .build();
         }
+
+        InputStream con = new ByteArrayInputStream(webAccess.getDocument().getBytes());
```

Comment:
Ich sehe den Kontext hier gerade nicht, aber auch hier ist der ByteArrayInputStream vermutlich überflüssig und Du kannst einfach den String weiterreichen.

Reply:
Da gibt es die Funktion ```readItems``` die mit JSON arbeitet, aber it's solved